### ISSUE TITLE
Columns: SortField and Filter params not passed to LazyDataModel

### ISSUE THREAD

**Issue by mafaul** [primefaces/primefaces#4765]
## 1) Environment
- PrimeFaces version: 7.0
- Application server + version: Payara 5.191
- Theme & Layout: Barcelona v 2.0.0
- Affected browsers: Chrome

## 2) Expected behavior
Filter values entered in globalFilter on column filters should be passed to the lazyDataModel like it did in PF6.2. Same with sortField.

## 3) Actual behavior
Sortfield: null. Filters: {null=[]}

## 4) Steps to reproduce
Enter a value to filter by on a lazy dataTable.
 Note that the values psecified in the datatable asstibutes (sortBy and filterBy) works, filtering after the initial table load has issues.

**Comment by tandraschko** [comment]
Please provide a example or steps to reproduce in the showcase.

**Comment by mafaul** [comment]
I will upload example soon. Working on it.

**Comment by mafaul** [comment]
I got it working by specifying a value for the "field" property on the columns.

Docs: "Name of the field to pass lazy load method for filtering and sorting. If not specified, filterBy-sortBy values are used to identify the field name."

We used to pass the field name (filterBy-sortBy values) like filterBy=#{dbColumnName}. This does not work on PF7. I do however like using the dedicated "field" property more.

**Comment by melloware** [comment]
Shouldn't this still work though?  So to me it sounds like a bug.

**Comment by mafaul** [comment]
PF 4.0 did not have the "field" property on columns, so we had to find a _hacky_ workaround, I don't see this as a bug. 
It is confusing that specifying a value for "field" as well as setting "filterable" to true does not enable the filter for a column, you still have to give a dummy value for filterBy.

**Comment by mafaul** [comment]
Columns highlighted in yellow are never set.
It is almost as if the var (subField) is null at the time that it is needed.
![image](https://user-images.githubusercontent.com/12007706/58962603-dee1ad00-87ab-11e9-8c01-b368b49af3bd.png)

**Comment by mafaul** [comment]
It appears that this issue only appears on p:columns

**Comment by mridgez59** [comment]
This still is an issue in PF 7.0. Any updates?

**Comment by mdzaebel** [comment]
So the issue should be opened again.

**Comment by tandraschko** [comment]
can you please check it with 8.0.RC2?

**Comment by martin654** [comment]
Hi,
Isn't this related with https://github.com/primefaces/primefaces/issues/5162 ? Field attribute works for sorting or filtering only when sortBy or filterBy attributes specified no matter what you type in them... I think that sortable="true" and filterable="true " should work with combination with filed="myField1Name". (tested with 8.0RC2)

**Comment by mafaul** [comment]
It appears to be resolved by #30 

Tested with 8.0 RC3

  **p:column filterBy map as received in LazyDataModel:** 

```
    { integer2=FilterMeta [
	filterField=integer2, 
	columnKey=form:recordList:j_idt6, filterByVE=/index.xhtml @33,104 
	filterBy="#{item.integer2}", 
	filterMatchMode=STARTS_WITH, 
	filterValue=5]
	}|#]
```
  
**p:columns (Dynamic Columns) filterBy map as received in LazyDataModel:**
```
  {integer=FilterMeta [
	filterField=integer, 
	columnKey=form:recordList:j_idt8:1, filterByVE=/index.xhtml @37,179 
	filterBy="#{item[col.columnName]}", 
	filterMatchMode=STARTS_WITH, 
	filterValue=5]
	}|#]
```

**Comment by melloware** [comment]
Thanks for testing and closing!
